src/main/java/net/glowstone/Explosion.java

```diff
@@ -327,13 +331,13 @@ private void playOutSoundAndParticles() {
         }
     }
 
-    private void playOutExplosion(GlowPlayer player, Iterable<BlockVector> blocks) {
+    private void playOutExplosion(GlowPlayer player, Iterable<Block> blocks) {
         Collection<Record> records = new ArrayList<>();
 
-        for (BlockVector block : blocks) {
-            byte x = (byte) (block.getBlockX() - location.getBlockX());
-            byte y = (byte) (block.getBlockY() - location.getBlockY());
-            byte z = (byte) (block.getBlockZ() - location.getBlockZ());
+        for (Block block : blocks) {
+            byte x = (byte) (block.getLocation().getBlockX() - (int) location.getX());
```

Comment:
As example, lets take X = -103.351. When casting to int, we get -103. The .getBlockX() returns, as expected, -104.
The client apparently needs the offsets to be calculated from the casted location.